STYLE.md

```diff
@@ -0,0 +1,81 @@
+
+# Ops Python style guide
+
+This is the Python style guide we use for the Ops project. However, it's also the style we're converging on for other projects maintained by the Charm Tech team.
+
+We use Ruff for formatting, and run our code through the Pyright type checker. We also try to follow [PEP 8](https://peps.python.org/pep-0008/), the official Python style guide. However, PEP 8 is fairly low-level, so in addition we've come up with the following style guidelines.
+
+Of course, this is just a start! We add to this list as things come up in code review and we make a team decision.
+
+
+**Table of contents:**
+
+* [Simplicity](#simplicity)
+* [Specific decisions](#specific-decisions)
+* [Docs and docstrings](#docs-and-docstrings)
+
+
+## Simplicity
+
+### Avoid nested comprehensions
+
+"Flat is better than nested."
+
+**Don't:**
+
+```python
+units = [units for app in model.apps for unit in app.units]
+
+for current in (
+	status for status in pebble.ServiceStatus if status is not pebble.ServiceStatus.ACTIVE
+):
+	...
+```
+
+**Do:**
+
+```python
+units = []
+for app in model.apps:
+	for unit in app.units:
+		units.append(unit)
+
+active_statuses = [status for status in pebble.ServiceStatus if status != pebble.ServiceStatus.ACTIVE]
+for current in active_statuses:
+	...
```

Comment:
```suggestion
units = []
for app in model.apps:
    for unit in app.units:
        units.append(unit)

active_statuses = [status for status in pebble.ServiceStatus if status != pebble.ServiceStatus.ACTIVE]
for current in active_statuses:
    ...
```

Reply:
Spaces not tabs, as PEP8 would say :laughing:.

Reply:
Good catch, thanks. Fixed.